pkg/operator/controller/controller_dns_configmap.go

```diff
@@ -237,3 +248,21 @@ func coreDNSLogLevel(dns *operatorv1.DNS) string {
 	}
 	return "class error"
 }
+
+func contains(upstreams []operatorv1.Upstream, upstream operatorv1.Upstream) bool {
+	for _, anUpstream := range upstreams {
+		if upstream.Type == operatorv1.SystemResolveConfType && anUpstream.Type == upstream.Type {
+			return true
+		}
+		if upstream.Type == operatorv1.NetworkResolverType {
+			if anUpstream.Type == upstream.Type {
+				if strings.EqualFold(anUpstream.Address, upstream.Address) {
+					if anUpstream.Port == upstream.Port {
```

Comment:
Should empty and "53" be treated as equal?  That is, should the following:
```yaml
    - address: 100.1.1.1
      port: 53
      type: Network
```
be considered a duplicate of the following?
```yaml
    - address: 100.1.1.1
      type: Network
```
It might be worth using go-cmp to simplify this logic (e.g., something like `daemonsetConfigChanged` and `cmpTerminationGracePeriodSeconds` in `controller_dns_daemonset.go`).

Reply:
Nice catch.
Change made